### PR TITLE
[15.0][ADD] l10n_fr_ecotaxe_product: manage ecotax on products

### PR DESCRIPTION
**Pull request by SilvioC2C**
Allows managing ecotax functionalities directly on products instead of templates.
This way, a variant can have an ecotax configuration that is different from its template's configuration.

Includes and depends on #329

### CODE REVIEW

**Review by ivantodorovich — APPROVED** [review]

**Review by ajaniszewska-dev — APPROVED** [review]

**Comment by ivantodorovich** [comment]
@SilvioC2C 
https://github.com/OCA/l10n-france/pull/329 has already been merged, could you rebase?

**Comment by SilvioC2C** [comment]
@ivantodorovich done

File: l10n_fr_ecotaxe_product/tests/test_invoice_ecotaxe.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+from odoo.tests.common import Form
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+from .common import get_ecotax_cls
+
+
+# Using TestProductCommon to access `init_invoice` already defined in there
+class TestProductEcotaxe(AccountTestInvoicingCommon):
```
Comment (ivantodorovich) [inline l10n_fr_ecotaxe_product/tests/test_invoice_ecotaxe.py:16]: I guess we need to tag it as post_install

```
2022-05-31 07:39:05,692 265 ERROR odoo unittest.suite: ERROR: setUpClass (odoo.addons.l10n_fr_ecotaxe_product.tests.test_invoice_ecotaxe.TestProductEcotaxe)
Traceback (most recent call last):
  File "/__w/l10n-france/l10n-france/l10n_fr_ecotaxe_product/tests/test_invoice_ecotaxe.py", line 18, in setUpClass
    super().setUpClass(chart_template_ref)
  File "/opt/odoo/addons/account/tests/common.py", line 31, in setUpClass
    assert 'post_install' in cls.test_tags, 'This test requires a CoA to be installed, it should be tagged "post_install"'
AssertionError: This test requires a CoA to be installed, it should be tagged "post_install"
```

**Review by ivantodorovich — CHANGES_REQUESTED** [review]
Check pre-commit too

**Review by ivantodorovich — APPROVED** [review]

**Pull request merged**